kube-client/src/config/file_config.rs

```diff
@@ -525,6 +541,60 @@ impl AuthInfo {
     }
 }
 
+/// Cluster stores information to connect Kubernetes cluster used with auth plugins
+/// that have `provideClusterInfo`` enabled.
+/// This is a copy of [`kube::config::Cluster`] with certificate_authority passed as bytes without the path.
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct ExecAuthCluster {
+    /// The address of the kubernetes cluster (https://hostname:port).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub server: Option<String>,
+    /// Skips the validity check for the server's certificate. This will make your HTTPS connections insecure.
+    #[serde(rename = "insecure-skip-tls-verify")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub insecure_skip_tls_verify: Option<bool>,
+    /// PEM-encoded certificate authority certificates. Overrides `certificate_authority`
+    #[serde(rename = "certificate-authority-data")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(with = "base64serde")]
+    pub certificate_authority_data: Option<Vec<u8>>,
+    /// URL to the proxy to be used for all requests.
+    #[serde(rename = "proxy-url")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub proxy_url: Option<String>,
```

Comment:
based on upstream, this is missing a `disable_compression` bool also, not sure if that's important, but if we are porting, feels like we ought to bring it over

Reply:
I noticed that, but I felt that it'd be weird importing it as we don't have it in the "original" cluster struct, and it will always be unused (unless we add it to the original cluster then pass it from there to this one)